CvGameCoreDLL/CvRhyes.h

```diff
@@ -11,10 +11,10 @@ typedef list<char*> LISTCHAR;
 
 #define MAX_COM_SHRINE			(20)
 
-#define GREATPALACE				(1)
-#define SUMMERPALACE			(1)
-//#define VERSAILLES				(2)
-#define FORBIDDENPALACE			(2)
+#define GREATPALACE				(GC.getInfoTypeForString("BUILDING_GREAT_PALACE"))

```

Comment:
We should not use getInfoTypeForString here ... defines do not store constants, they only do code replacement. In other words, the expensive string operation is called whenever the constant is queried which is not what we want.